prediction_market_agent/agents/think_thoroughly_agent/think_thoroughly_agent.py

```diff
@@ -68,46 +62,21 @@ class Scenarios(BaseModel):
     scenarios: list[str]
 
 
-class TavilySearchResultsThatWillThrow(TavilySearchResults):
-    @tenacity.retry(
-        stop=tenacity.stop_after_attempt(3),
-        wait=tenacity.wait_fixed(1),
-        reraise=True,
-    )
-    def _run(
-        self,
-        query: str,
-        run_manager: CallbackManagerForToolRun | None = None,
-    ) -> tuple[list[dict[str, str]] | str, dict[t.Hashable, t.Any]]:
-        """
-        Use the tool.
-        Throws an exception if it occurs, instead stringifying it.
-        """
-        raw_results = self.api_wrapper.raw_results(
-            query,
-            self.max_results,
-        )
-        return self.api_wrapper.clean_results(raw_results["results"]), raw_results
-
-    @tenacity.retry(
-        stop=tenacity.stop_after_attempt(3),
-        wait=tenacity.wait_fixed(1),
-        reraise=True,
-    )
-    async def _arun(
-        self,
-        query: str,
-        run_manager: AsyncCallbackManagerForToolRun | None = None,
-    ) -> tuple[list[dict[str, str]] | str, dict[t.Hashable, t.Any]]:
-        """
-        Use the tool asynchronously.
-        Throws an exception if it occurs, instead stringifying it.
-        """
-        raw_results = await self.api_wrapper.raw_results_async(
-            query,
-            self.max_results,
-        )
-        return self.api_wrapper.clean_results(raw_results["results"]), raw_results
+@tool
+@observe()
+def tavily_search_tool(query: str) -> list[dict[str, str]]:
```

Comment:
In the new CrewAI versions, they abandoned using Langchain tools directly, and they have their own "BaseTool" class, which needs to be subclassed.

They do have `crewai-tools` package, which offers Tavily, however is impossible to make it compatible with other packages we are using.

Anyway, this is a pretty simple and elegant solution.